File: players-service/src/main/java/es/ual/players_service/model/Player.java
package es.ual.players_service.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.LocalDate;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

@Entity
@Table(name = "players")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Player {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @Column(nullable = false)
  private String name;

  private String firstName;

  private String lastName;

  private Integer age;

  private LocalDate birthdate;

  private String nationality;

  private String height;

  private String weight;

  private Integer number;

  private String team;

  private String league;

  private String position;

  private String photo;

  @JdbcTypeCode(SqlTypes.JSON)
  @Column(columnDefinition = "jsonb")
  private Location location;

  private LocalDateTime created;
}

File: player-roster-app/src/utils/player.ts
export interface Player {
  id: string;
  name: string;
  firstName?: string;
  lastName?: string;
  age?: number;
  birthdate?: string;
  nationality?: string;
  height?: string;
  weight?: string;
  number?: number;
  team?: string;
  league?: string;
  position?: string;
  photo?: string;
  externalId?: number;
  location?: { type: 'Point'; coordinates: number[] };
  created?: string;
}

export async function fetchPlayers(apiBase: string): Promise<Player[]> {
  const res = await fetch(`${apiBase}/players`);
  if (!res.ok) throw new Error(`Error fetching players: ${res.statusText}`);
  return res.json();
}

export async function fetchPlayerById(apiBase: string, id: string): Promise<Player> {
  const res = await fetch(`${apiBase}/players/${id}`);
  if (!res.ok) throw new Error(`Error fetching player ${id}: ${res.statusText}`);
  return res.json();
}
